libraries/chain/block_log.cpp

```diff
@@ -197,6 +197,23 @@ namespace eosio { namespace chain {
          return bh;
       }
 
+      uint32_t get_block_num_at(fc::datastream<fc::cfile>& file, uint64_t position) {
+         // to derive blknum_offset==14 see block_header.hpp and note on disk struct is packed
+         //   block_timestamp_type timestamp;                  //bytes 0:3
+         //   account_name         producer;                   //bytes 4:11
+         //   uint16_t             confirmed;                  //bytes 12:13
+         //   block_id_type        previous;                   //bytes 14:45, low 4 bytes is big endian block number
+         //   of previous block
+
+         file.seek_end(0);
+         EOS_ASSERT(position <= file.tellp(), block_log_exception, "Invalid block position ${position}",
```

Comment:
This really should be:
```suggestion
         EOS_ASSERT(position +  blknum_offset + sizeof(uint32_t) <= file.tellp(), block_log_exception, "Invalid block position ${position}",
```

with `blknum_offset` declared before.

Reply:
This EOS_ASSERT was in the original function. Will correct it. Thanks.